components/service/glean/src/main/java/mozilla/components/service/glean/storages/GenericScalarStorageEngine.kt

```diff
@@ -177,6 +185,35 @@ internal abstract class GenericScalarStorageEngine<ScalarType> : StorageEngine {
         }
     }
 
+    /**
+     * Return a set of the all the labels in use for a given labeled metric.
+     * This includes labels in all of the pings specified by the metric, and
+     * for all lifetimes.
+     *
+     * @param metricData The labeled metric to find labels for.
+     * @return The set of labels currently in use.
+     */
+    @Suppress("NestedBlockDepth")
+    override fun getLabelsForMetric(metricData: CommonMetricData): Set<String> {
+        // Make sure data with "user" lifetime is loaded before getting the snapshot.
+        ensureUserLifetimeLoaded()
+
+        val prefix = "${metricData.identifier}/"
```

Comment:
Instead of doing this and creating a special case, is there any way we could use `extraSerializationData` for this purpose? If it's too tricky, then I'm fine with just moving ahead with the current approach.

Reply:
I'm not seeing a way, but I might be missing something.  Since labeled metrics don't have their own storage engine, we can't override `extraSerializationData`.

Reply:
Oh, good point. You know what? let's move forward with this and maybe file a bug to see if there's a better way (e.g. removing labelled info from leaking into the generic scalar storage engine)

Reply:
I thought of a way to make this feel less "special-casey" and not have understanding of labeled metrics in `GenericScalarStorageEngine`.  If `GenericScalarStorageEngine` just provided an iterator over all of the "names" in the store, `LabeledMetricType` could filter on the ones it cares about...

Reply:
Oh! That sounds smart, I like it! Do you prefer landing this as it is and working on that as a follow-up or just deal with it here?

Reply:
I'll probably just deal with that here and ping you when it's ready for another look.